pkg/provision/providers/qemu/arch.go

```diff
@@ -94,25 +93,43 @@ func (arch Arch) PFlash(uefiEnabled, secureBootEnabled bool, extraUEFISearchPath
 			return nil
 		}
 
-		uefiSourcePaths := []string{
-			"/usr/share/ovmf/OVMF.fd",
-			"/usr/share/OVMF/OVMF.fd",
-			"/usr/share/OVMF/OVMF_CODE_4M.fd",
-			"/usr/share/OVMF/OVMF_CODE_4M.secboot.fd",
+		// Default search paths
+		uefiSourcePathPrefixes := []string{
+			"/usr/share/ovmf",
+			"/usr/share/OVMF",
+			"/usr/share/qemu",
 		}
 
-		if _, err := os.Stat("/usr/share/qemu/edk2-x86_64-secure-code.fd"); err == nil {
-			// alpine uses this path
-			uefiSourcePaths = append(uefiSourcePaths, "/usr/share/qemu/edk2-x86_64-secure-code.fd")
+		// Secure boot enabled firmware files
+		uefiSourceFiles := []string{
+			"OVMF.secboot.fd",
+			"OVMF_CODE.secboot.fd",
+			"OVMF_CODE_4M.secboot.fd",
+			"edk2-x86_64-secure-code.fd", // Alpine Linux
 		}
 
+		// Non-secure boot firmware files
+		uefiSourceFilesInsecure := []string{
+			"OVMF.fd",
+			"OVMF_CODE.fd",
+			"OVMF_CODE_4M.fd",
+		}
+
+		if !secureBootEnabled {
+			uefiSourceFiles = append(uefiSourceFiles, uefiSourceFilesInsecure...)
+		}
+
+		// Append extra search paths
 		for _, p := range extraUEFISearchPaths {
-			uefiSourcePaths = append(uefiSourcePaths, filepath.Join(p, "OVMF.fd"))
+			uefiSourcePathPrefixes = append(uefiSourcePathPrefixes, p)
```

Comment:
```suggestion
			uefiSourcePathPrefixes = append(uefiSourcePathPrefixes, extraUEFISearchPaths...)
```

this would avoid the looping